google-cloud-app_engine-v1/proto_docs/google/appengine/v1/version.rb

```diff
@@ -0,0 +1,597 @@
+        #     Duration that static files should be cached by web proxies and browsers.
+        #     Only applicable if the corresponding
+        #     [StaticFilesHandler](https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1/apps.services.versions#StaticFilesHandler)
+        #     does not specify its own expiration time.
+        #
+        #     Only returned in `GET` requests if `view=FULL` is set.
+        # @!attribute [rw] health_check
+        #   @return [::Google::Cloud::AppEngine::V1::HealthCheck]
+        #     Configures health checking for instances. Unhealthy instances are
+        #     stopped and replaced with new instances.
+        #     Only applicable in the App Engine flexible environment.
+        #
+        #     Only returned in `GET` requests if `view=FULL` is set.
+        # @!attribute [rw] readiness_check
+        #   @return [::Google::Cloud::AppEngine::V1::ReadinessCheck]
+        #     Configures readiness health checking for instances.
+        #     Unhealthy instances are not put into the backend traffic rotation.
+        #
+        #     Only returned in `GET` requests if `view=FULL` is set.
+        # @!attribute [rw] liveness_check
+        #   @return [::Google::Cloud::AppEngine::V1::LivenessCheck]
+        #     Configures liveness health checking for instances.
+        #     Unhealthy instances are stopped and replaced with new instances
+        #
+        #     Only returned in `GET` requests if `view=FULL` is set.
+        # @!attribute [rw] nobuild_files_regex
+        #   @return [::String]
+        #     Files that match this pattern will not be built into this version.
+        #     Only applicable for Go runtimes.
+        #
+        #     Only returned in `GET` requests if `view=FULL` is set.
+        # @!attribute [rw] deployment
+        #   @return [::Google::Cloud::AppEngine::V1::Deployment]
+        #     Code and application artifacts that make up this version.
+        #
+        #     Only returned in `GET` requests if `view=FULL` is set.
+        # @!attribute [rw] version_url
+        #   @return [::String]
+        #     Serving URL for this version. Example:
+        #     "https://myversion-dot-myservice-dot-myapp.appspot.com"
+        # @!attribute [rw] endpoints_api_service
+        #   @return [::Google::Cloud::AppEngine::V1::EndpointsApiService]
+        #     Cloud Endpoints configuration.
+        #
+        #     If endpoints_api_service is set, the Cloud Endpoints Extensible Service
+        #     Proxy will be provided to serve the API implemented by the app.
+        # @!attribute [rw] entrypoint
+        #   @return [::Google::Cloud::AppEngine::V1::Entrypoint]
+        #     The entrypoint for the application.
+        # @!attribute [rw] vpc_access_connector
+        #   @return [::Google::Cloud::AppEngine::V1::VpcAccessConnector]
+        #     Enables VPC connectivity for standard apps.
+        class Version
```

Comment:
I just noticed how much documentation is placed in the `proto_docs/.../version.rb` file? Is this a mistake or is it necessary?

Reply:
Do we have a naming collision with `Version`?

Reply:
There's a 1:1 correspondence between input proto files and the output proto_docs files. It really shouldn't matter that much. These files are never loaded at runtime (i.e. they're not in lib/). They're present only for YARD.

As for `Version`, no there isn't a naming collision. The standard `VERSION` constant is all caps, so there isn't a risk of a proto type colliding with our generated library version.

It is conceivable that a client could have a service named `Version` which would cause a collision (not in the class name but in the file name). That would trigger a generator error because protoc doesn't like it if you try to create multiple files with the same name. We haven't encountered any cases of this so far. If/when we do, we'll need to put in some special case logic in the generator to handle it.